src/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -1,58 +1,388 @@
+  - `ChainSync` issues `Graphsync` requests to its peers randomly for the `Blocks` and `StateTree` for `LastTrustedCheckpoint`:
+    - `ChainSync`'s counterparts in other peers MUST provide the state tree.
+    - It is only semi-rational to do so, so `ChainSync` may have to try many peers.
+    - Some of these requests MAY fail.
+- **Chain State and Finality**:
+  - In this state, the **chain MUST NOT advance** beyond whatever the node already has.
+  - No new blocks are reported to consumers.
+  - The chain state provided is the available `Blocks` and `StateTree` for `LastTrustedCheckpoint`.
+- **Important Notes:**
+  - `ChainSync` needs to fetch several blocks: the `Block` pointed at by `LastTrustedCheckpoint`, and its direct `Block.Parents`.
+  - Nodes only need hashing to validate these `Blocks` and `StateTrees` -- no block validation or state machine computation is needed.
+  - The initial value of `LastTrustedCheckpoint` is `GenesisCheckpoint`, but it MAY be a value later in Chain history.
+  - `LastTrustedCheckpoint` enables efficient syncing by making the implicit economic consensus of chain history explicit.
+  - By allowing fetching of the `StateTree` of `LastTrustedCheckpoint` via `Graphsync`, `ChainSync` can yield much more
+    efficient syncing than comparable blockchain synchronization protocols, as syncing and validation can start there.
+  - Nodes DO NOT need to validate the chain from `GenesisCheckpoint`. `LastTrustedCheckpoint` MAY be a value later in Chain history.
+  - Nodes DO NOT need to but MAY sync earlier `StateTrees` than `LastTrustedCheckpoint` as well.
+- Pseudocode 1: a basic version of `SYNC_CHECKPOINT`:
+    ```go
+    func (c *ChainSync) SyncCheckpoint() {
+        while !c.HasCompleteStateTreeFor(c.LastTrustedCheckpoint) {
+            selector := ipldselector.SelectAll(c.LastTrustedCheckpoint)
+            c.Graphsync.Pull(c.Peers, sel, c.IpldStore)
+            // Pull SHOULD NOT pull what c.IpldStore already has (check first)
+            // Pull SHOULD pull from different peers simultaneously
+            // Pull SHOULD be efficient (try different parts of the tree from many peers)
+            // Graphsync implementations may not offer these features. These features
+            // can be implemented on top of a graphsync that only pulls from a single
+            // peer and does not check local store first.
+        }
+        c.ChainCatchup() // on to CHAIN_CATCHUP
+    }
+    ```
+- **security conditions to transition out:**
+  - `StateTree` for `LastTrustedCheckpoint` MUST be stored locally and verified (hashing is enough)
+- **transitions out:**
+  - once node receives and verifies complete `StateTree` for `LastTrustedCheckpoint`: move to `CHAIN_CATCHUP`
+
+## ChainSync FSM: `CHAIN_CATCHUP`
+
+- While in this state:
+  - `ChainSync` is well-bootstrapped, and has an initial **trusted** `StateTree` to start from.
+  - `ChainSync` is receiving latest `Blocks` from `BlockPubsub`
```

Comment:
Why does it need this? This seems irrelevant to fetching and validating up to `TargetHeads`, or do you explicitly mean that `TargetHeads` should be changing? If the latter, completely closing the gap between `ChainSync.FinalityTipset ... ChainSync.BestTargetHead` could be very difficult as the target keeps moving, but also unnecessary before switching to "follow" mode so long as validation has reached FinalityTipset.